contrib/roles/linux/kubernetes/tasks/generate_certs_master.yml

```diff
@@ -88,6 +88,7 @@
 
 - name: Kubernetes Certs | Generating Certificates
   shell: |
+    set -o errexit
```

Comment:
This might be redundant since the make-certs already set it. Is there any case in which the ``cd`` command would fail here and not when creating the folder at the beginning of the file?

Reply:
Not really, but I feel it's just best practice to have shell scripts with proper error handling.

There are just some paranoia casses that I can think of. For example: this yaml gets updated in the future, and this task might have an empty `{{ kubernetes_certificates.tmp_generate_path }}` variable, due to various changes to the playbook. The `cd` command will fail, but the script will continue and potentially execute some random `./make-certs` script from the current working directory (if randomly present there).

Anyhow, I agree with you that this is really less likely to become a problem.

But I feel that we should at least add a `&&` between commands, just for the sake of proper error handling for our shell scripts in the Ansible playbooks. What do you think ?

Reply:
It will fail anyway when creating the directory. But both methods work, let's just go with ``set -o errexit`` everywhere to have consistency and it's also useful for multi-line commands.